### PR TITLE
Fix warnings with new GCCs and glibcs

### PR DESCRIPTION
**Pull request by dm0-**
This fixes warnings when building with recent glibc versions.

Part of coreos/coreos-overlay#2886

### CODE REVIEW

File: src/update_engine/filesystem_iterator.cc [hunk 1]
```diff
@@ -111,11 +111,10 @@ void FilesystemIterator::Increment() {
 void FilesystemIterator::IncrementInternal() {
   CHECK_EQ(dirs_.size(), names_.size() + 1);
   for (;;) {
-    struct dirent dir_entry;
     struct dirent* dir_entry_pointer;
-    int r;
+    errno = 0;
     RETURN_ERROR_IF_FALSE(
-        (r = readdir_r(dirs_.back(), &dir_entry, &dir_entry_pointer)) == 0);
+        (dir_entry_pointer = readdir(dirs_.back())) != NULL || errno != 0);
```
Comment (lucab) [inline src/update_engine/filesystem_iterator.cc]: I may be misreading this, but I think the check should be `p != NULL || errno == 0`. The error condition we are trying to catch is the `p == NULL && errno != 0`, but negated.

Comment (dm0-) [reply src/update_engine/filesystem_iterator.cc]: Yes, thanks.  Updated.

**Comment by bgilbert** [comment]
It looks as though gtest provides pkg-config files now?

**Comment by dm0-** [comment]
If it does, Gentoo's latest version doesn't install them.

**Pull request merged**